### PR TITLE
Update deps and add MultiThreading analyzer

### PR DESCRIPTION
**Pull request by Bond-009**
`SmartAnalyzers.MultithreadingAnalyzer` prevents potential issues related to multithreading.
Ref: - [Avoid multithreading traps with Roslyn: Lock object selection](https://cezarypiatek.github.io/post/avoid-multithreading-traps-p1/)
- [Avoid thread synchronization problems with Roslyn: Synchronization primitives traps](https://cezarypiatek.github.io/post/avoid-multithreading-traps-p2/)

### CODE REVIEW

File: Emby.Server.Implementations/AppBase/BaseConfigurationManager.cs [hunk 1]
```diff
@@ -98,16 +98,31 @@ protected BaseConfigurationManager(IApplicationPaths applicationPaths, ILoggerFa
...
+                }
+
+                lock (_configurationSyncLock)
+                {
+                    if (_configurationLoaded)
+                    {
+                        return _configuration;
+                    }
+
+                    _configuration = (BaseApplicationConfiguration)ConfigurationHelper.GetXmlConfiguration(ConfigurationType, CommonApplicationPaths.SystemConfigurationFilePath, XmlSerializer);
```
Comment (JustAMan) [inline Emby.Server.Implementations/AppBase/BaseConfigurationManager.cs]: are you telling me C# doesn't have some well-defined Singleton support, so you had to hand-roll it?..

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 2]
```diff
@@ -106,6 +85,31 @@ public class InstallationManager : IInstallationManager
...
+        public event EventHandler<InstallationEventArgs> PackageInstalling;
+
+        public event EventHandler<InstallationEventArgs> PackageInstallationCompleted;
+
+        public event EventHandler<InstallationFailedEventArgs> PackageInstallationFailed;
+
+        public event EventHandler<InstallationEventArgs> PackageInstallationCancelled;
+
+        /// <summary>
+        /// Occurs when [plugin uninstalled].
```
Comment (JustAMan) [inline Emby.Server.Implementations/Updates/InstallationManager.cs]:
```suggestion
        /// Occurs when a plugin is uninstalled.
```

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 3]
```diff
@@ -106,6 +85,31 @@ public class InstallationManager : IInstallationManager
...
+        /// </summary>
+        public event EventHandler<GenericEventArgs<IPlugin>> PluginUninstalled;
+
+        /// <summary>
+        /// Occurs when [plugin updated].
```
Comment (JustAMan) [inline Emby.Server.Implementations/Updates/InstallationManager.cs]:
```suggestion
        /// Occurs when a plugin is updated.
```

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 4]
```diff
@@ -106,6 +85,31 @@ public class InstallationManager : IInstallationManager
...
+        /// </summary>
+        public event EventHandler<GenericEventArgs<(IPlugin, PackageVersionInfo)>> PluginUpdated;
+
+        /// <summary>
+        /// Occurs when [plugin updated].
```
Comment (JustAMan) [inline Emby.Server.Implementations/Updates/InstallationManager.cs]:
```suggestion
        /// Occurs when a plugin is installed.
```

File: Emby.Server.Implementations/AppBase/BaseConfigurationManager.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/AppBase/BaseConfigurationManager.cs]: ? This just ensures the loading of the config is thread safe.

Comment (JustAMan) [reply Emby.Server.Implementations/AppBase/BaseConfigurationManager.cs]: Okay, I checked and C# has a few ways of making a singleton but nothing completely built-in or fitting, let's close this conversation on that.

**Comment by JustAMan** [comment]
@Bond-009 why are suddenly so many checks falling? Are we expecting public ABI to break?

**Pull request merged**